logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java

```
@@ -112,32 +209,66 @@
  * @author Ceki G&uuml;lc&uuml;
  * @author S&eacute;bastien Pennec
  */
-public class RequestLogImpl extends ContextBase implements RequestLog, AppenderAttachable<IAccessEvent>, FilterAttachable<IAccessEvent> {
-
+public class RequestLogImpl extends ContextBase implements RequestLog, LifeCycle, AppenderAttachable<IAccessEvent>, FilterAttachable<IAccessEvent> {
     public final static String DEFAULT_CONFIG_FILE = "etc" + File.separatorChar + "logback-access.xml";
 
+    enum State {
+        FAILED, STOPPED, STARTING, STARTED, STOPPING
+    }
+    State state = State.STOPPED;
+
     AppenderAttachableImpl<IAccessEvent> aai = new AppenderAttachableImpl<IAccessEvent>();
     FilterAttachableImpl<IAccessEvent> fai = new FilterAttachableImpl<IAccessEvent>();
     String fileName;
     String resource;
-    boolean started = false;
+
+    // Default to non-modern Jetty (ie: Jetty 9.3 or earlier)
+    // Jetty 9.4.x and newer is considered modern.
+    boolean modernJettyRequestLog = false;
     boolean quiet = false;
 
     public RequestLogImpl() {
         putObject(CoreConstants.EVALUATOR_MAP, new HashMap<String, EventEvaluator<?>>());
+
+        // plumb the depths of Jetty and the environment ...
+        try {
+            Class.forName("jakarta.servlet.http.HttpServlet");
+            throw new RuntimeException("The new jakarta.servlet classes are not supported by this " +
+                "version of logback-access (check for a newer version of logback-access that " +
+                "does support it)");
+        } catch(Throwable ignore) {
+            // Class doesn't exist, assumption is that this is normal javax.servlet environment.
+        }
```

This `catch` will swallow the `Exception` you are trying to raise here...